resize.js

```diff
@@ -0,0 +1,76 @@
+var preserve_ratio = "%(preserve_ratio)s"; // string
+var min_height = %(min_height)d;
+var min_width = %(min_width)d;
+var resizable_style = "%(resizable_style)s";
+
+function resizeImage(idx, img){
+    $resizeImage($(img));
+}
+
+async function $resizeImage($img){
+    var img = $img[0];
+    while (img.naturalWidth == 0 || img.naturalHeight == 0) {
+        await new Promise(r => setTimeout(r, 1000));
+    }
+    var preserve_ratio_in_resizable = false;
+    if (preserve_ratio == "current" || preserve_ratio == "original"){
+        preserve_ratio_in_resizable = true;
+    }
+    if ($img.resizable("instance") == undefined ) {
+        $maybe_remove_a_dimension($img);
+        minHeight = ((min_height == null) ? 0: min_height)
+        minWidth = ((min_width == null) ? 0: min_width)
+        $img.resizable({
+            aspectRatio: preserve_ratio_in_resizable,
+            minHeight: minHeight,
+            minWidth: minWidth
+        });
+        $img.css("max-width", "100%%"); //%% because a single percent would make a python error during formatting of this file.
+        $img.dblclick(onDblClick);
+        $divUi = $img.parents("div[class^=ui-]");
+        $divUi.attr("contentEditable", "false");
+        $divUi.css("display", "inline-block");
+    } else {
+        console.log("Trying to apply resizable to image already resizable.");
+    }
+}
+
+function onDblClick(){
+    img = this;
+    $img = $(img);
+    $img.css("width", "");
+    $img.css("height", "");
+    $parents = $img.parents("div[class^=ui-]");
+    $parents.css("width", "");
+    $parents.css("height", "");
+}
+
+function $cleanResize($field){
+    // clean everything related to resize, so that it can be saved and
+    // displayed properly in reviewer
+    $divUi = $field.find("div[class^=ui-]");
+    $divUi.replaceWith(
+        function() {
+            return $(this).contents();
+        }
+    );
+    $field.find("img").each(partialCleanResize);
+}
+
+function partialCleanResize(idx, img){
+    $partialCleanResize($(img));
+}
+
+function $partialCleanResize($img){
```

Comment:
Can ask why this is needed? From what I can see, this removes those stylings that the user may have applied and need. For example, I know that Anking uses max-height to control how the images are displayed in AnkiMobile and AnkiDesktop.

Reply:
One of the problem there is with "jquery-ui's resizable" is that it adds a lot of styling information to images. Ideally, I would have hoped that applying jquery-ui's resizable's destruct method would remove everything added by jquery-ui. Alas, that's not the case at all. So I have to remove information explictly before saving the images.
I guess I could save all previous values of each style and put them back during saving if required.